api/base/serializers.py

```diff
@@ -69,14 +75,27 @@ def to_internal_value(self, data):
 
 
 class TypeField(ser.CharField):
+    """
+    Type field that validates that 'type' in the request body is the same as the Meta type.
+
+    Also ensures that type is write-only and required.
+    """
     def __init__(self, **kwargs):
         kwargs['write_only'] = True
         kwargs['required'] = True
         super(TypeField, self).__init__(**kwargs)
 
+    # Overrides CharField
     def to_internal_value(self, data):
-        if self.root.Meta.type_ != data:
+        request_data = self.context['request'].data
+        if isinstance(request_data, list):
+            meta = self.root.child.Meta.type_
```

Comment:
self.root would be the ListSerializer, self.root.child would be the NodeSerializer for example.  The list serializer does not have Meta information.  I wanted the TypeField to be able to handle if the request was single or bulk.